### DISCUSSION TITLE
CLAP Audio Plugin + Nuked SC-55 Support

### DISCUSSION THREAD

**Discussion by nickkeane** [86Box/86Box#5054] [Ideas]
DOSBox staging has recently developed a PR for support for a CLAP audio plugin system and Nuked SC-55.

https://github.com/dosbox-staging/dosbox-staging/pull/4090

**Comment by Kappa971** [comment]
This appears to have been implemented: https://github.com/86Box/86Box/pull/7138
The problem is that it seems like a ghost implementation, undocumented. I've only figured out that you need to create a "plugins" folder inside "roms" with the .clap file, but I have no idea where the ROMs are supposed to be placed.

**Reply by ucucanok** [reply]
Instructions as well as the required plug-in and roms can be found [here](https://github.com/johnnovak/Nuked-SC55-CLAP)

**Reply by Kappa971** [reply]
I managed to get it working. The .clap file goes in `roms\plugins`, while the roms go in `roms\plugins\Nuked-SC55-Resources\ROMs\SC-55-v1.21` (or depending on the rom version) with bin file naming as indicated on that site.

However, this feature remains hidden, and many people probably don't know that it's possible to emulate an SC-55 in 86Box without external software. It's not even mentioned in the changelog or official documentation: https://86box.readthedocs.io/en/latest/settings/sound.html
Is it just an oversight or is there a reason to make it so hidden?

**Reply by lemondrops** [reply]
> The .clap file goes in roms\plugins, while the roms go in roms\plugins\Nuked-SC55-Resources\ROMs\SC-55-v1.21

Or in `C:\Program Files\Common Files\CLAP\` or `%LOCALAPPDATA%\Programs\CLAP\` as mentioned on that page.

> Is it just an oversight or is there a reason to make it so hidden?

I believe the documentation has not been fully updated yet.

**Reply by shermp** [reply]
Another option (what I added to DOSBox Staging) is for the emulator to set the `SOUNDCANVAS_ROM_PATH` environment variable before loading the CLAP plugin. This will allow lookup in "standard" 86Box ROM directories, or provide a GUI to allow the user to specify their own ROM path.

**Comment by Kappa971** [comment]
The Nuked SC-55 license has been modified. I think it is now possible to implement SC-55 emulation directly into 86Box, without the need for external plugins.
